### PR TITLE
Dp 1186 2

### PR DESCRIPTION
**Pull request by JordanBalesBAO**
### Related Issues
- DP-XXX
- etc.

## Reviewer:
- [ ] Tests added/updated & passing.
- [ ] Clean linting.
- [ ] Related issue ticket in Jira/GitHub.
- [ ] Documentation added/updated.
- [ ] Code conforms to style guidelines.
- [ ] Well commented.
- [ ] Updates reflected in NEWS.md.
- [ ] Build check passes.

### CODE REVIEW

**Comment by JordanBalesBAO** [comment]
So I updated the unit test to mostly only look at 2024. There are approximately 6 that are failing and I would like assistance looking at.

File: R/checkAnalytics.R [hunk 1]
```diff
@@ -791,6 +663,7 @@ checkAnalytics <- function(d,
     dplyr::ungroup()
 
   #Special analytics indicators needed for some checksin COP23
+  #NOTE did not update this for COP24, as it seemed to be specific? Need to confirm
```
Comment (jason-p-pickering) [inline R/checkAnalytics.R:666]: Please confirm.

File: R/writePSNUxIM.R [hunk 2]
```diff
@@ -133,107 +133,18 @@ writePSNUxIM <- function(d,
...
-                     d2_session = d2_session)
-
-    d$tool$wb <- r$wb
-    d$info$messages <- appendMessage(d$info$messages, r$info$messages$message, r$info$messages$level)
-    d$info$newSNUxIM <- TRUE
-
-  }
-
-  if (d$info$cop_year == 2023) {
+  # if (d$info$cop_year == 2023) {
```
Comment (jason-p-pickering) [inline R/writePSNUxIM.R:136]: Remove commented code.

File: R/writePSNUxIM.R [hunk 3]
```diff
@@ -294,7 +205,13 @@ writePSNUxIM <- function(d,
       dplyr::filter(!is.na(org_type)) %>%
       dplyr::select(dp_label, orgUnit = uid)
 
-    schema <- cop23_psnuxim_schema
+    schema <- pick_schema(d$info$cop_year, "PSNUxIM")
+    # schema <- if (d$info$cop_year == 2023) {
```
Comment (jason-p-pickering) [inline R/writePSNUxIM.R:209]: Remove commented code.

File: R/writePSNUxIM.R [hunk 4]
```diff
@@ -133,107 +133,18 @@ writePSNUxIM <- function(d,
...
 
     d$info$has_psnuxim <- !is.null(d$data$SNUxIM)
 
     dp_datim_map <- getMapDataPack_DATIM_DEs_COCs(cop_year = d$info$cop_year)
     targets_data <- prepareTargetsData(d, append)
-    template_file <- system.file("extdata", "COP23_PSNUxIM_Template.xlsx", package = "datapackr")
-
+    template_file <- system.file("extdata",
+                                 paste0("COP",
+                                        d$info$cop_year %% 100,
```
Comment (jason-p-pickering) [inline R/writePSNUxIM.R:144]: I think this will work for now, but it's not totally intuitive.

**Review by jason-p-pickering — CHANGES_REQUESTED** [review]
Have requested a few small changes. In general, I think we can proceed once these small issues are fixed.

File: R/checkAnalytics.R (hunk 1 above)
Comment (JordanBalesBAO) [reply R/checkAnalytics.R]: Confirmed with Christian and added it

**Pull request merged**